packages/backend-common/src/scm/git.ts

```diff
@@ -86,13 +86,14 @@ export class Git {
     return git.commit({ fs, dir, message, author, committer });
   }
 
-  async clone({ url, dir }: { url: string; dir: string }): Promise<void> {
+  async clone({ url, dir, ref }: { url: string; dir: string; ref?: string }): Promise<void> {
     this.config.logger?.info(`Cloning repo {dir=${dir},url=${url}}`);
     return git.clone({
       fs,
       http,
       url,
       dir,
+      ref,
```

Comment:
https://isomorphic-git.org/docs/en/clone.html

We should probably not provide `ref` if it is undefined.

Reply:
@OrkoHunter this seems to indicate that sending ref=undefined should do the right thing:

https://github.com/isomorphic-git/isomorphic-git/blob/b102e1d9a8f48ec20042be5ff0feb9961007e904/src/commands/clone.js#L90

```js
  ref = ref || defaultBranch
```

Because otherwise I don't know how to do this other than an if-then-else with different function calls.